src/geb/trans.lisp

```diff
@@ -50,8 +50,20 @@
                                        (poly:+ (poly:* cz xin)
                                                (poly:- yzin cy)
                                                (poly:* cx cy)))))))
-    (pair          (error "not implemented"))
-    (case          (let ((obj-to-nat ))))
+    (pair          (let* ((z  (codom (mcdr obj)))
+                          (cz (obj-to-nat z)))
+                     (poly:* cz (poly:+ (to-poly (mcar obj))
+                                        (to-poly (mcdr obj))))))
```

Comment:
This should match

```idris
substMorphToBNC (SMPair {x} {y} {z} f g) with (substObjToNat y, substObjToNat z)
  substMorphToBNC (SMPair {x} {y} {z} f g) | (cy, cz) =
    #| cz #* substMorphToBNC f #+ substMorphToBNC g
```

Reply:
> So in CL presentation we do not have the constant constructor for BNCPolyM?

Correct, the sum type just has `fixnum` as a variant, and thus it all types out. The power of anonymous sum types a gogo!